text/3579-pki.md

```diff
@@ -0,0 +1,309 @@
+
+These individuals should be available in the event quorum is needed for root key operations. These roles can and should be reappointed as needed, but to simplify logistics, these roles should not require rotation more often than 2-3 years. (Operations requiring quorum are expected to be rare, and an annual rotation would make rotation of the quorum group the vast majority of quorum operations.)
+
+##### When the Quorum will be needed
+[when-the-quorum-will-be-needed]: #when-the-quorum-will-be-needed
+
+- Changes to quorum membership or thresholds shall require a quorum (enforced by the HSM authentication)
+- Replacement of the Root Key in the event of expiration or compromise
+- Creation, signing, and issuance of a new Delegate Key
+- Creation, signing, and issuance of a replacement Delegate Key in the event of expiration or compromise
+
+##### Quorum Threat Mitigations
+
+- To reduce the feasibility of coercion or collusion, we recommend that no more than 3 (M-2) of the 9 people be affiliated with any one company or institution, or be residents or citizens of the same country. (If in the future the quorum threshold is changed, this limit should be changed accordingly to remain M-2).
+- If in the future a desire arises to modify the quorum, we strongly recommend growing the size of membership rather than reducing the quorum threshold. The quorum exists to mitigate against bad actors, and is not implemented to prove a majority.
+- To reduce the feasibility of coercion or collusion, we recommend that members of the quorum be selected from within the Rust project, community, Foundation staff and/or Foundation Board. This allows us to spread trust and responsibility among socially and fiduciarily responsible parties.
+- There is value in the members of the quorum generally residing in geographically and politically distributed locations most of the time. This RFC does not place any requirements regarding quorum member colocation for events or travel; the project and the quorum members can appropriately evaluate the value of such events and the corresponding risks.
+- Hardware keys shall be utilized for two factor authentication for Quorum IAM accounts; phone-based or weaker methods of 2FA authentication cannot be used
+
+### Delegate Keys
+
+Delegate keys shall be used for purposes within the realms of their designated responsibility and team(s). It shall be up to the individual implementors using these keys to make sure to verify their trust up to the Rust root or to the appropriate delegate key for their usage. For example, it shall be an implementation detail for the crates.io and cargo teams on whether to verify signatures up to the Rust root key, or to their specific delegate keys for other purposes. This is done to allow for the greatest flexibility of users of the keys.
```

Comment:
While I can see the desire to avoid committing to this right now, in the future there needs to be coordination between owners of delegate keys whose verifiers go up to the shared root. That needs to happen before any delegate keys are certified, lest confusion between different uses could arise (for example, a validly signed bors commit being confused as a validly signed crate).

I think this deserves a note of caution, and a suggestion that it is preferable to root things at the delegated keys where possible to avoid the need for technically constraining verifiers. If that is doable, such uses don't need to wait for a central PKI to be deployed, or really depend on it at all (see the earlier discussions on whether rust benefits from a central PKI at all).